File: solutions/fsharp/line-up/1/LineUp.fs
module LineUp

let format (name: string) (number: int): string =
    let suffix =
        let lastTwo = number % 100
        if lastTwo >= 11 && lastTwo <= 13 then
            "th"
        else
            match number % 10 with
            | 1 -> "st"
            | 2 -> "nd"
            | 3 -> "rd"
            | _ -> "th"

    $"{name}, you are the {number}{suffix} customer we serve today. Thank you!"

File: solutions/php/line-up/1/LineUp.php
<?php declare(strict_types=1);

function format(string $name, int $number): string
{
    $suffix = match ($number % 100) {
        11, 12, 13 => 'th',
        default => match ($number % 10) {
            1 => 'st',
            2 => 'nd',
            3 => 'rd',
            default => 'th',
        }
    };
    return "{$name}, you are the {$number}{$suffix} customer we serve today. Thank you!";
}